### PR TITLE
(x) ORCOMP-293 Added Filter builder support for enum values

### PR DESCRIPTION
**Pull request by alexfdezsauco**
Fixes #ORCOMP-293

### Checklist

- [x] I have included examples or tests
- [x] I am listed in the CONTRIBUTORS file
- [x] I have cleaned up the commit history (use rebase and squash)

### Changes proposed in this pull request:

- Added Filter builder support for enum values

### CODE REVIEW

File: src/Orc.FilterBuilder.Tests/Orc.FilterBuilder.Tests.Shared/Expressions/EnumExpressionFacts.cs [hunk 1]
```diff
@@ -0,0 +1,393 @@
+﻿using System;
```
Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder.Tests/Orc.FilterBuilder.Tests.Shared/Expressions/EnumExpressionFacts.cs]: Add file header, use R# CTRL E + F

File: src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/EnumExpression.cs [hunk 2]
```diff
@@ -0,0 +1,112 @@
+namespace Orc.FilterBuilder
```
Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/EnumExpression.cs]: Add file header

File: src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/EnumExpression.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+    using Catel;
+    using Catel.Reflection;
+    using Models;
+
+    [DebuggerDisplay("{ValueControlType} {SelectedCondition} {Value}")]
+    public class EnumExpression<TEnum> : NullableDataTypeExpression where TEnum : struct
```
Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/EnumExpression.cs]: where TEnum : struct on new line

File: src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/EnumExpression.cs [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    {
+        #region Constructors
+        /// <summary>
+        /// Initialize an instance of <see cref="EnumExpression{TEnum}"/>.
+        /// </summary>
+        /// <param name="isNulleable">Inicates wheter the type is nulleable</param>
```
Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/EnumExpression.cs]: "Indicates whether the type is nullable."

Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/EnumExpression.cs]: "Indicates whether the type is nullable."

File: src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/PropertyExpression.cs [hunk 5]
```diff
@@ -51,125 +51,88 @@ private void OnPropertyChanged()
 
             if (Property != null)
             {
-                if (Property.Type.IsEnumEx())
+                var propertyType = Property.Type;
+                bool isNulleable = propertyType.IsNullableType();
```
Comment (GeertvanHorrik) [inline src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Expressions/PropertyExpression.cs]: isNullable

**Comment by alexfdezsauco** [comment]
Fixed

**Pull request merged**